app/models/git/exercise.rb

```diff
@@ -73,24 +93,38 @@ def solution_files
 
     # This includes meta files
     memoize
-    def non_ignored_files
-      non_ignored_filepaths.each.with_object({}) do |filepath, hash|
+    def tooling_files
+      tooling_filepaths.each.with_object({}) do |filepath, hash|
         hash[filepath] = read_file_blob(filepath)
       end
     end
 
     # This includes meta files
     memoize
-    def non_ignored_filepaths
+    def tooling_filepaths
       filepaths.select do |filepath| # rubocop:disable Style/InverseMethods
         !filepath.match?(track.ignore_regexp)
       end
     end
 
     # This includes meta files
     memoize
-    def non_ignored_absolute_filepaths
-      non_ignored_filepaths.map { |filepath| full_filepath(filepath) }
+    def tooling_absolute_filepaths
+      tooling_filepaths.map { |filepath| full_filepath(filepath) }
+    end
+
+    memoize
+    def cli_filepaths
+      special_filepaths = ['README.md', 'HELP.md']
+      special_filepaths << 'HINTS.md' if filepaths.include?('.docs/hints.md')
+
+      filtered_filepaths = filepaths.select do |filepath| # rubocop:disable Style/InverseMethods
+        !filepath.match?(track.ignore_regexp) &&
+          !filepath.start_with?('.docs/') &&
+          !filepath.start_with?('.meta/')
```

Comment:
I've hardcoded the excluding of files in the `.docs` and `.meta` directory, but let me know if that is too course. I could do individual files.

Reply:
No, I think that's correct.

I'd probably rather guard, than daisy chain logic though:

e.g.

```ruby
filtered_filepaths = filepaths.select do |filepath|
  next if filepath.match?(ignore_regexp)
  next if filepath.start_with?('.docs/')
  next if filepath.start_with?('.meta/')

  true
end
```

---

What did we decide about the ignore regexp? Are we keeping it? I can't remember.

Reply:
We decided we'd remove it. I wanted to do that later though, or would you like me to remove it in this PR?

Reply:
No, we can take that as a second step. Maybe add a `# TODO` in this Pr for it to remind us?

Reply:
Done